src/types.js

```diff
@@ -86,8 +86,19 @@ export type PmRecipientUser = {
   short_name: string,
 };
 
+/**
+ * Indexer over all narrows for which we have locally stored messages.
+ * @prop narrow - Map from narrows to lists of message IDs.
+ */
 export type NarrowsState = {
-  [narrow: string]: Message[],
+  [narrow: string]: number[],
+};
+
+/**
+ * Indexer over all locally stored messages.
```

Comment:
Thanks for adding these jsdoc comments! We should do more of these. :smile: 

The term "indexer" doesn't mean much to me here; I'll make a pass post-merge at trying to come up with another way to describe these.

(The one sense that comes to mind is in the context of a search engine's backend, where the "indexer" is a throughput-optimized system that collates giant quantities of data to produce/maintain the "index", which in turn is the data structure that the latency-optimized "server" consults to answer a query. E.g. [livegrep](https://github.com/livegrep/livegrep/blob/master/src/indexer.cc) uses this term, as does anybody that's worked at Google. But that's clearly not the sense you mean.)

Reply:
The Flow error that's in CI right now shows me where Flow uses this term in a different way! Perhaps this is what you had in mind.

```
 • Message [1] is incompatible with array type [2] in the indexer property.
 • number [3] is incompatible with string [4] in the indexer property's key.
...
 [4][2]  93│   const messagesState: { [narrow: string]: Message[] } =
```

Or at least the term "indexer property". By which it seems to mean this kind of pseudo-property/meta-property that describes the object functioning like a map.

As a term for the object itself (an object that has an "indexer property" and no others), I'd just call it a "map".

Reply:
`Indexer` is actually Flow terminology, it's explained here: https://flow.org/en/docs/types/objects/#toc-objects-as-maps

Reply:
Ah, your comment was quicker :) Yeah, `map` sounds good as well, probably better in the sense that one doesn't need to know Flow for it.